src/nnet3/nnet-optimize.h

```diff
@@ -76,29 +77,50 @@ void Optimize(const NnetOptimizeOptions &config,
               const ComputationRequest &request,
               NnetComputation *computation);
 
-
+struct ComputationRequestHasher {
+  size_t operator()(const ComputationRequest *cr) const {
+    size_t ans = 0;
+    ans = cr->inputs.size() * kPrime + cr->outputs.size();
+    ans += (size_t)cr;
+    return ans;
+  }
+ private:
+  static const int kPrime = 7853;
+};
+ 
 /// This class enables you to do the compilation and optimization in one call,
 /// and also ensures that if the ComputationRequest is identical to the previous
 /// one, the compilation process is not repeated.
 class CachingOptimizingCompiler {
  public:
-  CachingOptimizingCompiler(const Nnet &nnet): nnet_(nnet) { }
+  CachingOptimizingCompiler(const Nnet &nnet,
+                            const int32 capacity = 20):
```

Comment:
You should make capacity part of the configuration class, and give it a more informative name, e.g. num-computations-cached.